reccmp/isledecomp/compare/core.py

```diff
@@ -262,51 +273,82 @@ def orig_bin_checker(addr: int) -> bool:
+            for fun in codebase.iter_name_functions():
+                batch.set_orig(
+                    fun.offset, type=EntityType.FUNCTION, stub=fun.should_skip()
                 )
-            else:
-                self._db.match_variable(var.offset, var.name)
 
-        for tbl in codebase.iter_vtables():
-            self._db.match_vtable(tbl.offset, tbl.name, tbl.base_class)
+                if fun.name.startswith("?"):
+                    batch.set_orig(fun.offset, symbol=fun.name)
+                else:
+                    batch.set_orig(fun.offset, name=fun.name)
+
+            for var in codebase.iter_variables():
+                batch.set_orig(var.offset, name=var.name, type=EntityType.DATA)
+                if var.is_static and var.parent_function is not None:
+                    batch.set_orig(
+                        var.offset, static_var=True, parent_function=var.parent_function
+                    )
 
-        for string in codebase.iter_strings():
-            # Not that we don't trust you, but we're checking the string
-            # annotation to make sure it is accurate.
-            try:
-                # TODO: would presumably fail for wchar_t strings
-                orig = self.orig_bin.read_string(string.offset).decode("latin1")
-                string_correct = string.name == orig
-            except UnicodeDecodeError:
-                string_correct = False
-
-            if not string_correct:
-                logger.error(
-                    "Data at 0x%x does not match string %s",
+            for tbl in codebase.iter_vtables():
+                batch.set_orig(
+                    tbl.offset,
+                    name=tbl.name,
+                    base_class=tbl.base_class,
+                    type=EntityType.VTABLE,
+                )
+
+        # For now, just redirect match alerts to the logger.
+        report = create_logging_wrapper(logger)
+
+        # Now match
+        match_symbols(self._db, report)
+        match_functions(self._db, report)
+        match_vtables(self._db, report)
+        match_static_variables(self._db, report)
+        match_variables(self._db, report)
+
+        with self._db.batch() as batch:
+            for string in codebase.iter_strings():
+                # Not that we don't trust you, but we're checking the string
+                # annotation to make sure it is accurate.
+                try:
+                    # TODO: would presumably fail for wchar_t strings
+                    orig = self.orig_bin.read_string(string.offset).decode("latin1")
+                    string_correct = string.name == orig
+                except UnicodeDecodeError:
+                    string_correct = False
+
+                if not string_correct:
+                    logger.error(
+                        "Data at 0x%x does not match string %s",
+                        string.offset,
+                        repr(string.name),
+                    )
```

Comment:
Shouldn't this be a report? Maybe it also makes sense to go over all remaining log calls and see which of those should be reports.